src/napari/layers/shapes/shapes.py

```diff
@@ -1259,14 +1257,15 @@ def z_index(self, z_index):
             self._data_view.update_z_index(i, z_idx)
 
     @property
-    def selected_data(self):
-        """set: set of currently selected shapes."""
+    def selected_data(self) -> Selection[int]:
+        """Selection: set of currently selected shapes."""
         return self._selected_data
 
     @selected_data.setter
-    def selected_data(self, selected_data):
-        self._selected_data = set(selected_data)
-        self._selected_box = self.interaction_box(self._selected_data)
+    def selected_data(self, selected_data: Sequence[int]) -> None:
+        self._selected_data.clear()
+        self._selected_data.update(Selection(selected_data))
```

Comment:
```suggestion
        self._selected_data.update(selected_data)
```

Why do you need to create a mid object?

Reply:
I think I missunderstood the way the `Selection` container worked - i.e., that in order for `layer.selected` data to return a `Selection`, the private variable `_selected_data` would also have to be of that type 🤔